controller/src/controller.rs

```diff
@@ -105,6 +122,10 @@ impl<T: BottlerocketNodeClient> BrupopController<T> {
         None
     }
 
+    fn init_metrics(&mut self) {
+        let meter = global::meter("brupop-controller");
+        self.metrics = Some(BrupopControllerMetrics::new(meter));
+    }
```

Comment:
I tried it before. However it only works when I init meter inside run() method, neither parse meter as argument nor initialize in the constructor works.

Reply:
Can you elaborate on what went wrong? I don't understand why it wouldn't work.

Reply:
It does have some global state (obviously), but not sure why it would cause a problem. https://github.com/open-telemetry/opentelemetry-rust/blob/5d6d000c47f9f5da307e32c559bf43c40a08f951/opentelemetry/src/global/metrics.rs#L6

Reply:
The differences are when init meter inside `run()`, the meter is 
```
{"v":0,"name":"brupop-controller","msg":"[NEW - END]","level":30,"hostname":"brupop-controller-deployment-54cb49c8c6-flsrt","pid":1,"time":"2022-01-12T18:41:24.716404272+00:00","target":"controller::metrics","line":45,"file":"controller/src/metrics.rs","meter":"Meter { instrumentation_library: InstrumentationLibrary { name: \"brupop-controller\", version: None }, core: UniqueInstrumentMeterCore { inner: Accumulator(AccumulatorCore { current: {}, async_instruments: Mutex { data: AsyncInstrumentState { runners: [], instruments: [] }, poisoned: false, .. }, current_epoch: AtomicNumber(0), processor: BasicProcessor { aggregator_selector: Histogram([0.5, 0.9, 0.99]), export_selector: Cumulative, state: Mutex { data: BasicProcessorState { config: BasicProcessorConfig { memory: true }, values: {}, process_start: SystemTime { tv_sec: 1642012884, tv_nsec: 715485442 }, interval_start: SystemTime { tv_sec: 1642012884, tv_nsec: 715485513 }, interval_end: SystemTime { tv_sec: 1642012884, tv_nsec: 715485573 }, started_collection: 0, finished_collection: 0 }, poisoned: false, .. } }, resource: Resource { attrs: {} } }), sync_state: Mutex { data: {}, poisoned: false, .. }, async_state: Mutex { data: {}, poisoned: false, .. } } }","elapsed_milliseconds":0}
```

Whereas when init meter in the constructor/main the meter is 
```
{"v":0,"name":"brupop-controller","msg":"[NEW - START]","level":30,"hostname":"brupop-controller-deployment-54cb49c8c6-njh5m","pid":1,"time":"2022-01-12T19:05:54.096074979+00:00","target":"controller::metrics","line":48,"file":"controller/src/metrics.rs","meter":"Meter { instrumentation_library: InstrumentationLibrary { name: \"brupop-controller\", version: None }, core: NoopMeterCore { _private: () } }"}
```

Reply:
I don't find an good example on how to set up for `UniqueInstrumentMeterCore` manually,  it seems can be done via the [registry](https://github.com/open-telemetry/opentelemetry-rust/blob/ccc8fb210c046856825e283940850478521c9c8b/opentelemetry/src/metrics/registry.rs) , but only example I found [here](https://github.com/open-telemetry/opentelemetry-rust/blob/02ddf1b9eac1ead48510504fc6bd4cdccb7e23fb/opentelemetry/src/sdk/metrics/controllers/push.rs#L174-L182)  seems overkilled.

Reply:
I'm having a hard time working out why this would happen.

Reply:
Found out why, comments in latest change.